graphdatascience/query_runner/protocol/retry_utils.py

```diff
@@ -0,0 +1,43 @@
+import logging
+import signal
+import typing
+from types import FrameType
+from typing import Optional
+
+from tenacity import RetryCallState, retry_base
+
+
+class retry_unless_signal(retry_base):
+    """Retries unless one of the given signals is raised."""
+
+    def __init__(self, signals: list[signal.Signals]) -> None:
```

Comment:
should pass a terminationFlag here (which is based on signals) from the main thread